MekHQ/src/mekhq/campaign/finances/Loan.java

```diff
@@ -93,102 +85,79 @@ public Loan(Money p, int r, int c, int y, int s, GregorianCalendar today, String
     }
 
     public void setFirstPaymentDate() {
-        boolean keepGoing = true;
         //We are going to assume a standard grace period, so you have to go
         //through the first full time length (not partial) before your first
         //payment
-        int grace = 1;
-        switch(schedule) {
-        case Finances.SCHEDULE_BIWEEKLY:
-            grace = 2;
-            break;
-        case Finances.SCHEDULE_QUARTERLY:
-            grace = 3;
-            break;
-        }
-        while(keepGoing) {
-            nextPayment.add(GregorianCalendar.DAY_OF_YEAR, 1);
-            switch(schedule) {
+
+        // First, we need to increase the number of days by one
+        nextPayment = nextPayment.plusDays(1);
+
+        // Then, we create a variable to hold temporary difference values
+        int temp;
+
+        // Finally, we use that and the schedule type to determine the length including the grace period
+        switch (schedule) {
             case Finances.SCHEDULE_BIWEEKLY:
-                if(nextPayment.get(Calendar.DAY_OF_WEEK) == 1) {
-                    if(grace > 0) {
-                        grace--;
-                    }
-                    else {
-                        keepGoing = false;
-                    }
+                temp = 8 - nextPayment.getDayOfWeek().getValue();
```

Comment:
This logic needs an explanation, if you're just looking for the next Monday/Sunday or something like that there is a more obvious way:
```
d.with(TemporalAdjusters.next(DayOfWeek.MONDAY));
```